### ISSUE TITLE
Examples from PEP 747 (`TypeForm`) throw errors

### ISSUE THREAD

**Issue by rijenkii** [microsoft/pyright#10001]
**Describe the bug**
[Examples from PEP 747](https://peps.python.org/pep-0747/#specification) do not pass the type checker. Specifically, examples `ok1`, `ok3`, `ok5`, `ok6` and `ok7` throw errors.

**Code or Screenshots**
```python
from typing_extensions import TypeForm, Literal, Optional, Any


ok1: TypeForm[str | None] = str | None
ok2: TypeForm[str | None] = str
ok3: TypeForm[str | None] = None
ok4: TypeForm[str | None] = Literal[None]
ok5: TypeForm[str | None] = Optional[str]
ok6: TypeForm[str | None] = "str | None"
ok7: TypeForm[str | None] = Any
```

```
> PYRIGHT_PYTHON_FORCE_VERSION=1.1.395 pyright py.py
/tmp/asd/py.py
  /tmp/asd/py.py:4:29 - error: Type "UnionType" is not assignable to declared type "TypeForm[str | None]"
    "UnionType" is not assignable to "TypeForm[str | None]" (reportAssignmentType)
  /tmp/asd/py.py:6:29 - error: Type "None" is not assignable to declared type "TypeForm[str | None]"
    "None" is not assignable to "TypeForm[str | None]" (reportAssignmentType)
  /tmp/asd/py.py:8:29 - error: Type "UnionType" is not assignable to declared type "TypeForm[str | None]"
    "UnionType" is not assignable to "TypeForm[str | None]" (reportAssignmentType)
  /tmp/asd/py.py:9:29 - error: Type "Literal['str | None']" is not assignable to declared type "TypeForm[str | None]"
    "Literal['str | None']" is not assignable to "TypeForm[str | None]" (reportAssignmentType)
  /tmp/asd/py.py:10:29 - error: Type "type[Any]" is not assignable to declared type "TypeForm[str | None]"
    Type "Any" is not assignable to type "str | None"
      "Any" is not assignable to "str"
      "Any" is not assignable to "None" (reportAssignmentType)
5 errors, 0 warnings, 0 informations 
```

**VS Code extension or command-line**
Version 1.1.395, CLI.

**Comment by erictraut** [comment]
PEP 747 is not yet approved, so support for it in pyright is experimental. If you want to make use of it, you need to set `enableExperimentalFeatures` to true in your pyright config.
